src/jit/codegenxarch.cpp

```diff
@@ -5000,12 +5001,14 @@ void CodeGen::genConsumePutStructArgStk(GenTreePutArgStk* putArgNode, regNumber
             GenTreeLclVarCommon* lclNode = src->AsLclVarCommon();
 
             // Generate LEA instruction to load the LclVar address in RSI.
-            getEmitter()->emitIns_R_S(INS_lea, emitTypeSize(src), srcReg, lclNode->gtLclNum, 0);
+            // Source is a local (on the stack). Use EA_PTRSIZE.
+            getEmitter()->emitIns_R_S(INS_lea, EA_PTRSIZE, srcReg, lclNode->gtLclNum, 0);
         }
         else
         {
             assert(src->gtRegNum != REG_NA);
-            getEmitter()->emitIns_R_R(INS_mov, emitTypeSize(src), srcReg, src->gtRegNum);
+            // Source is not a local (not on the stack). Use EA_BYREF.
```

Comment:
I think that the type used for the src register should be the same here and below - since an EA_BYREF will be tracked as a byref register. It may not be necessary (i.e. `gcMarkRegSetNpt` is called in any case, and doesn't care whether or howit was being tracked, it will clear it from both sets), but I think it's the right thing to do for consistency.